File: src/main/java/net/treset/minecraftlauncher/util/CreationStatus.java
package net.treset.minecraftlauncher.util;

import net.treset.mc_version_loader.files.DownloadStatus;
import net.treset.minecraftlauncher.LauncherApplication;

public class CreationStatus {
    private final DownloadStep currentStep;
    private final DownloadStatus downloadStatus;

    public CreationStatus(DownloadStep currentStep, DownloadStatus downloadStatus) {
        this.currentStep = currentStep;
        this.downloadStatus = downloadStatus;
    }

    public DownloadStep getCurrentStep() {
        return currentStep;
    }

    public DownloadStatus getDownloadStatus() {
        return downloadStatus;
    }

    public enum DownloadStep {
        STARTING("creator.status.starting"),
        MODS("creator.status.mods"),
        OPTIONS("creator.status.options"),
        RESOURCEPACKS("creator.status.resourcepacks"),
        SAVES("creator.status.saves"),
        VERSION("creator.status.version"),
        VERSION_VANILLA("creator.status.version.vanilla"),
        VERSION_ASSETS("creator.status.version.assets"),
        VERSION_LIBRARIES("creator.status.version.libraries"),
        VERSION_FABRIC("creator.status.version.fabric"),
        JAVA("creator.status.java"),
        FINISHING("creator.status.finishing");
        private final String translationKey;
        DownloadStep(String translationKey) {
            this.translationKey = translationKey;
        }

        public String getMessage() {
            return LauncherApplication.stringLocalizer.get(translationKey);
        }
    }
}

File: client/composeApp/src/commonMain/kotlin/net/treset/treelauncher/backend/util/CreationStatus.kt
package net.treset.treelauncher.backend.util

import dev.treset.mcdl.util.DownloadStatus
import net.treset.treelauncher.localization.strings

data class CreationStatus(val currentStep: DownloadStep, val downloadStatus: DownloadStatus?) {

    enum class DownloadStep(val message: () -> String) {
        STARTING({ strings().creator.status.starting() }),
        MODS({ strings().creator.status.mods() }),
        OPTIONS({ strings().creator.status.options() }),
        RESOURCEPACKS({ strings().creator.status.resourcepacks() }),
        SAVES({ strings().creator.status.saves() }),
        VERSION({ strings().creator.status.version.value() }),
        VERSION_VANILLA({ strings().creator.status.version.vanilla() }),
        VERSION_ASSETS({ strings().creator.status.version.assets() }),
        VERSION_LIBRARIES({ strings().creator.status.version.libraries() }),
        VERSION_FILE({ strings().creator.status.version.file() }),
        VERSION_FABRIC({ strings().creator.status.version.fabric() }),
        VERSION_FABRIC_LIBRARIES({ strings().creator.status.version.fabricLibraries() }),
        VERSION_FABRIC_FILE({ strings().creator.status.version.fabricFile() }),
        VERSION_FORGE({ strings().creator.status.version.forge() }),
        VERSION_FORGE_LIBRARIES({ strings().creator.status.version.forgeLibraries() }),
        VERSION_FORGE_FILE({ strings().creator.status.version.forgeFile() }),
        VERSION_QUILT({ strings().creator.status.version.quilt() }),
        VERSION_QUILT_LIBRARIES({ strings().creator.status.version.quiltLibraries() }),
        JAVA({ strings().creator.status.java() }),
        FINISHING({ strings().creator.status.finishing() });
    }
}
